pos_warning_exiting/static/src/js/chrome.js

```diff
@@ -0,0 +1,87 @@
+odoo.define("pos_warning_exiting.Chrome", function (require) {
+    "use strict";
+
+    const {_t} = require("web.core");
+    const Chrome = require("point_of_sale.Chrome");
+    const Registries = require("point_of_sale.Registries");
+    const {isRpcError} = require("point_of_sale.utils");
+
+    var _super_chromemodel = Chrome.prototype;
+
+    const PosWarningChrome = (Chrome) =>
+        class extends Chrome {
+            // @Override
+            async _closePos() {
+                var self = this;
+                const draft_orders_description = [];
+                const order_jsons = this.env.pos.db.get_unpaid_orders();
+
+                if (order_jsons.length) {
+                    order_jsons.forEach(function (order_json) {
+                        if (
+                            order_json.pos_session_id === self.env.pos.pos_session.id &&
+                            order_json.lines.length
+                        ) {
+                            var order_description =
+                                _t("Order #") + order_json.sequence_number;
+                            // Lazy depedency to pos_restaurant
+                            // to display table information
+                            if (order_json.table) {
+                                order_description +=
+                                    _t(" - Table: ") + order_json.table.name;
+                            }
+                            draft_orders_description.push(order_description);
+                        }
+                    });
+                }
```

Comment:
Not blocking here but I would do something like this here

```js
const unpaidOrders = this.env.pos.db.get_unpaid_orders();
const toLoseOrders = unpaidOrders.filter(
    order => order.pos_session_id === this.env.pos.session_id.id && order.lines.length
);
if (toLoseOrders.length) {
    const orderDescriptions = toLoseOrders.map(order => {
        let description = this.env._t("Order #") + order.sequence_number;
        // Lazy dependency to pos_restaurant to display table information
        if (order.table) {
            description += this.env._t(" - Table: ") + order.table.name;
        }
        return description;
    })
    // .. you know the rest
}
```

by using arrow functions you avoid having to define and use `self`, and you achieve a more readable result IMO